heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp

```diff
@@ -186,6 +190,27 @@ void StMgrClientMgr::HandleStMgrClientRegistered() {
   }
 }
 
+void StMgrClientMgr::HandleStMgrClientReconnect(const sp_string &_stmgr_id) {
+  // The default client reconnect time is 1s and the default max attempt is 300.
+  // Therefore if a connection is broken, stmgr would be restartd in 5 minutes by stmgr client.
+  // Although we want to restart stmgr sooner if there are more broken connections,
+  // it could be risky if the threshold is too agressive and causing stmgr to restart too soon.
+  // Here the max attempt across all clients is set to: per client max attempt * client count / 8.
+  // Hence if all connections are broken, the lower bound of restart time is about 40s.
+  // However in reality, broken connections cause extra delays hence less reconnects are made.
+  // In one case, it takes more than 120 minutes for a stmgr client to reach the max attempt and
+  // restart. So the stmgr is expected to be restarted in 10 to 20 minutes with the / 8 factor.
+  // The mininum max attempt is per client max attempt * 2 in case number of clients is too low.
```

Comment:
how is your proposal different from 'change 300 to 300/8'

Reply:
The issue here is for a bad stmgr, all connections are failing and it is more likely to be a local problem. with a global max attempt, stmgr doesn't need to wait per client failure count to reach 300. Because network failure can be very slow, bad stmgr tends to restart a lot slower than good stmgr.

when there is only one bad connection, it is more likely to be a remote problem, so we don't  want the stmgr to kill itself too soon. changing 300 to 300/8 makes good stmgrs to quit a lot sooner and this is not what we want.

Reply:
cool, got it

Reply:
The PR is a simple demo and I don't really like this solution myself. I just think we need a check based on global view. Any ideas are welcome.